applications/StructuralMechanicsApplication/custom_advanced_constitutive/multi_linear_isotropic_plane_stress_2d.cpp

```diff
@@ -0,0 +1,172 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:         BSD License
+//                   license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Klaus B. Sautter
+//
+// System includes
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "includes/checks.h"
+#include "includes/properties.h"
+#include "custom_advanced_constitutive/multi_linear_isotropic_plane_stress_2d.h"
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+//******************************CONSTRUCTOR*******************************************
+//************************************************************************************
+
+MultiLinearIsotropicPlaneStress2D::MultiLinearIsotropicPlaneStress2D()
+    : LinearPlaneStress()
+{
+}
+
+//******************************COPY CONSTRUCTOR**************************************
+//************************************************************************************
+
+MultiLinearIsotropicPlaneStress2D::MultiLinearIsotropicPlaneStress2D(const MultiLinearIsotropicPlaneStress2D& rOther)
+    : LinearPlaneStress(rOther)
+{
+}
+
+//********************************CLONE***********************************************
+//************************************************************************************
+
+ConstitutiveLaw::Pointer MultiLinearIsotropicPlaneStress2D::Clone() const
+{
+    MultiLinearIsotropicPlaneStress2D::Pointer p_clone(new MultiLinearIsotropicPlaneStress2D(*this));
+    return p_clone;
+}
+
+//*******************************DESTRUCTOR*******************************************
+//************************************************************************************
+
+MultiLinearIsotropicPlaneStress2D::~MultiLinearIsotropicPlaneStress2D()
+{
+}
+
+
+//************************************************************************************
+//************************************************************************************
+
+void MultiLinearIsotropicPlaneStress2D::CalculateElasticMatrix(Matrix& C, ConstitutiveLaw::Parameters& rValues)
+{
+    this->CheckClearElasticMatrix(C);
+
+    const Properties& r_material_properties = rValues.GetMaterialProperties();
+    const double nu = r_material_properties[POISSON_RATIO];
+
+    Vector current_strain = ZeroVector(this->VoigtSize);
```

Comment:
If the size is fixed could be interesting to use BoundedVector right?

Reply:
seems like I can't pass a BoundedVector to `GetStrainVector`, at least that's what my compiler tells me